rails/find_controller_action_from_uri.md

```diff
@@ -0,0 +1,28 @@
+# Find a controller and action from a URI string 
+
+Got `GET /posts`, want to craft a `redirect_to` based off of that URI and some other parameters?
+
+Today I learned that you can extract out the controller and action for a particular route with the magical 
+[`ActionDispatch::Routing::RouteSet#recognize_path`](http://www.rubydoc.info/docs/rails/ActionDispatch/Routing/RouteSet:recognize_path),
+which can be called through `Rails.application.routes.recognize_path`.
+
+So given a routes file:
+
+```ruby
+# config/routes.rb
+# ...
+resources :posts
+# ...
+```
+And jumping into the rails console:
+```bash
+$ bundle exec rails console
+```
+Try out the following:
+```ruby
+Rails.application.routes.recognize_path('/posts') #=> { controller: 'posts', action: 'index' }
+```
+
+I discovered this method via a failed refactoring, so I honestly don't have a great use-case for `#recognize_path`.
+
+Nevertheless, I think `#recognize_path` is an interesting, undocumented, and [huge](https://github.com/rails/rails/blob/v4.1.8/actionpack/lib/action_dispatch/routing/route_set.rb#L681-L719) part `ActionDispatch::Routing::RouteSet`'s public API!
```

Comment:
If something is undocumented, it is not part of the public API. `RouteSet` has no public API, as it is not part of the public API. I'm looking into why it's showing up on `api.rubyonrails.org` right now, but it's unintentional.